src/test/ui/const-generics/min_const_generics/const_fn_in_generics.rs

```diff
@@ -0,0 +1,19 @@
+// run-pass
+
+#![feature(min_const_generics)]
+#![allow(dead_code)]
+
+
+const fn identity<const T: u32>() -> u32 { T }
+
+pub struct ConstU32<const U: u32>;
+
+impl<const N: u32> ConstU32<N> {
+  pub fn new() -> ConstU32<{ identity::<3>() }> {
+    ConstU32::<{ identity::<3>() }>
+  }
+}
+
+fn main() {
+  let _v = ConstU32::<3>::new();
```

Comment:
Why did you expect `ConstU32::new()` to work? What would be the expected value of `N` in this case?

Reply:
Oops, I realize that the turbofish was on the impl block in order to access `ConstU32::new()`, but `ConstU32::new()` doesn't depend on N at all, and it returns a specific const value.

I assumed it was asking for it for the return value of new, but this behaviour seems to be line up with types so I was mistaken